lib/slurm_squeue_client.rb

```diff
@@ -61,7 +61,49 @@ def sinfo
     
     s.success? ? @sinfo = o : raise(CommandFailed, e)
   end
-  
+
+  # Parse and return total number of GPU nodes in a SLURM cluster.
+  # @return [Integer] number of GPU nodes
+  def gpu_nodes
+    return @available_gpu_nodes if defined?(@available_gpu_nodes)
+
+    Open3.pipeline_rw "sinfo -N -h --states=allocated,idle --Format='nodehost,gres:100'", 'uniq', 'grep gpu:v', 'wc -l' do |stdin, stdout|
```

Comment:
If you want all nodes, drop --states.  That will give you all possible GPU nodes.

Reply:
Make `gres:100` dynamic by finding longest gres and adding 10 by executing `sinfo -o '%G'` per other comments.